docs/api_specs/includes/watcher_api_specs.md

```diff
@@ -0,0 +1,650 @@
+</aside>
+
+### Byzantine events
+All of the following events indicate byzantine behaviour and that the user should either exit or challenge.
+
+#### `invalid_exit`
+> An invalid_exit event
+
+```json
+{
+    "event": "invalid_exit",
+    "details": {
+        "eth_height"  : 3521678,
+        "utxo_pos"  : 10001001,
+        "owner"  : "B3256026863EB6AE5B06FA396AB09069784EA8EA",
+        "currency"  : "0000000000000000000000000000000000000000",
+        "amount" : 100
+    }
+}
+```
+
+Indicates that an invalid exit is occurring. It should be challenged.
+
+
+#### `unchallenged_exit`
+> An unchallenged_exit event
+
+```json
+{
+    "event": "unchallenged_exit",
+    "details": {
+        "eth_height"  : 3521678,
+        "utxo_pos"  : 10001001,
+        "owner"  : "B3256026863EB6AE5B06FA396AB09069784EA8EA",
+        "currency"  : "0000000000000000000000000000000000000000",
+        "amount" : 100
+    }
+}
+```
+
+Indicates that an invalid exit is dangerously close to finalization and hasn't been challenged. User should exit.
+
+
+#### `invalid_block`
+> An invalid_block event
+
+```json
+{
+    "event": "invalid_block",
+    "details": {
+        "blockhash"  : "DB32876CC6F26E96B9291682F3AF4A04C2AA2269747839F14F1A8C529CF90225",
+        "blocknum"  : 10000,
+    }
+}
+```
+
+An invalid block has been added to the chain. User should exit.
+
+
+#### `block_withholding`
+> A block_withholding event
+
+```json
+{
+    "event": "block_withholding",
+    "details": {
+        "blockhash"  : "DB32876CC6F26E96B9291682F3AF4A04C2AA2269747839F14F1A8C529CF90225",
+        "blocknum"  : 10000,
+    }
+}
+```
+
+The ChildChain is withholding a block whose hash has been published on the root chain. User should exit.
+
+#### `noncanonical_ife`
+> A noncanonical_ife event
+
+```json
+{
+    "event": "noncanonical_ife",
+    "details": {
+        "txbytes": "F3170101C0940000..."
+    }
+}
+```
+
+An in-flight exit of a non-canonical transaction has been started. It should be challenged. 
+<aside class="warning"> Not Implemented Yet.</aside> 
+
+Event details:
+Attribute | Type | Description
+--------- | ------- | -----------
+txbytes | Hex encoded string | The in-flight transaction that the event relates to
+
+#### `invalid_ife_challenge`
+> A invalid_ife_challenge event
+
+```json
+{
+    "event": "invalid_ife_challenge",
+    "details": {
+        "txbytes": "F3170101C0940000..."
+    }
+}
+```
+
+A canonical in-flight exit has been challenged. The challenge should be responded to. 
+<aside class="warning"> Not Implemented Yet.</aside> 
+
+Event details:
+Attribute | Type | Description
+--------- | ------- | -----------
+txbytes | Hex encoded string | The in-flight transaction that the event relates to
+
+#### `piggyback_available`
+> A piggyback_available event
+
+```json
+{
+    "event": "piggyback_available",
```

Comment:
The input and output addresses are already in the `inflight_txs`, so the idea is that the user can look them up there, instead of having them duplicated here.